packages/gafl-webapp-service/src/processors/find-permit.js

```diff
@@ -1,8 +1,34 @@
 import filterPermits from './filter-permits.js'
 import crypto from 'crypto'
 import db from 'debug'
+
 const debug = db('webapp:find-permit')
 
+const addHashAndPermit = async (permission, request) => {
+  // To calculate a permit, hash and save
+
+  const { hash: _hash, permit: _permit, licensee: _licensee, ...hashOperand } = permission
+  const permit = await filterPermits(permission)
+  const updatedPermission = { ...permission, permit }
+
+  if (permit) {
+    if (!permit.newCostStartDate || !permit.newCost) {
+      debug('permit missing new cost details', updatedPermission)
+    }
+  } else {
+    debug("permit wasn't retrieved", updatedPermission)
+  }
+
+  const hash = crypto.createHash('sha256')
+  hash.update(JSON.stringify(hashOperand))
+  const updatedHash = hash.digest('hex')
+  const updatedRequest = await request.cache().helpers.transaction.setCurrentPermission({
+    ...updatedPermission,
+    hash: updatedHash
+  })
+  return updatedRequest
+}
```

Comment:
What you're doing here is barely changed from what happened originally. What addHashAndPermit is doing is still complex and wrong.
There should be a function that will retrieve a permit for the permission and return it to the caller.
Then there should be another function that will create an updated hash for a permission and return it to the caller.
Neither function should be saving anything. The name doesn't suggest that this is what happens, and in any case this should be done by the calling function so that it maintains control of its data.